core/src/main/java/org/apache/accumulo/core/metadata/schema/LinkingIterator.java

```diff
@@ -132,7 +146,23 @@ public TabletMetadata next() {
 
       if (prevTablet == null) {
         if (tmp.sawPrevEndRow()) {
-          currTablet = tmp;
+
+          Text prevRow = tmp.getExtent().getPrevEndRow();
+          Text prevMetaRow = null;
+          if (prevRow != null) {
+            prevMetaRow = TabletsSection.getRow(tmp.getExtent().getTableId(), prevRow);
+          }
+
+          // If the first tablet seen has a prev endrow within the range it means a preceding tablet
+          // exists that we need to go back and read. This could be caused by the first tablet in
+          // the range splitting concurrently while this code is reading.
+
+          if (prevRow == null || range.beforeStartKey(new Key(prevMetaRow))) {
```

Comment:
Could depend on nullness of prevMetaRow, so this line makes sense on its own without looking back at lines 152-154 where prevMetaRow's nullness is dependent on prevRow's nullness.